src/chords_generation/daria.py

```diff
@@ -0,0 +1,126 @@
+from typing import List, Tuple
+
+import music
+import utils.constants
+from . import daria_params as params
+
+# https://www.fim.uni-passau.de/fileadmin/dokumente/fakultaeten/fim/lehrstuhl/sauer/geyer/BA_MA_Arbeiten/BA-HausnerChristoph-201409.pdf
+'''
+Split notes into timeline
+Split timeline into windows (bars)
+    Extend some bars to prevent changing chord in the middle of sound
+For each bar get best chord using scoring lists
+'''
+
+
+def tonation_chords(tonation: music.Tonation):
+    n = tonation.note
+    if tonation.kind == "major":
+        chords = [
+            music.Chord(note=n, kind="major"),
+            music.Chord(note=n+2, kind="minor"),
+            music.Chord(note=n+4, kind="major"),
+            music.Chord(note=n+5, kind="major"),
+            music.Chord(note=n+7, kind="major7"),
+            music.Chord(note=n+9, kind="minor"),
+            music.Chord(note=n+11, kind="diminished")]
+    else:
+        chords = [
+            music.Chord(note=n, kind="minor"),
+            music.Chord(note=n+2, kind="diminished"),
+            music.Chord(note=n+4, kind="major"),
+            music.Chord(note=n+5, kind="minor"),
+            music.Chord(note=n+7, kind="major7"),
+            music.Chord(note=n+9, kind="major"),
+            music.Chord(note=n+11, kind="diminished")]
+    for i in range(len(chords)):
+        chords[i].level = i+1
+    return chords
+
+
+def points_for_sound(chord: music.Chord, sound: music.Sound):
+    if sound.note is None:
+        return 0
+    diff = (sound.note+12 - chord.note) % 12
+    return params.POINTS[chord.kind][diff]
+
+
+def point_coef(chord: music.Chord, first_sound: music.Sound, last_chord_level: int):
+    if chord.note == first_sound.note:
+        first_sound_coef = params.FIRST_SOUND
+    else:
+        first_sound_coef = 1
+    return params.COEFS[chord.kind] * \
+        params.NEXT_CHORD[last_chord_level].get(chord.level, 1) * \
```

Comment:
Move this default value = 1 to NEXT_CHORD declaration, as it won't be used with any other default value. You can do that after the 78th line in daria_params.py

Reply:
Ok